src/uu/cp/src/cp.rs

```diff
@@ -1138,7 +1138,8 @@ fn copy(sources: &[Source], target: &TargetSlice, options: &Options) -> CopyResu
     let mut hard_links: Vec<(String, u64)> = vec![];
 
     let mut non_fatal_errors = false;
-    let mut seen_sources = HashSet::with_capacity(sources.len());
+    let mut seen_sources = HashMap::with_capacity(sources.len());
+    let mut should_hard_linked_files = HashMap::with_capacity(sources.len());
```

Comment:
`seen_sources` is a name I understand immediately, but I don't really understand `should_hard_linked_files`? Can we give it a more descriptive name?